src/openrct2/ride/Vehicle.cpp

```diff
@@ -9213,7 +9213,22 @@ void Vehicle::InvalidateWindow()
 
 void Vehicle::UpdateCrossings() const
 {
-    if (TrainHead() != this)
+    auto curRide = GetRide();
+    if (curRide == nullptr)
+    {
+        return;
+    }
+
+    // Many New Element parks have invisible rides hacked into the path.
```

Comment:
This is not limited to parks from New Elements. Most RCT communities do things like this.

```suggestion
    // Parks may have invisible rides hacked into the path.
```